### PR TITLE
Security: introduce Utils::matchToken helper

### PR DESCRIPTION
**Pull request by macbre**
[PLATFORM-1703](https://wikia-inc.atlassian.net/browse/PLATFORM-1703)

Use `Wikia\Security\Utils::matchToken` for checking custom tokens in `Special:EmailTest`.

@wladekb / @Grunny

### CODE REVIEW

**Comment by wladekb** [comment]
:+1:

**Comment by Grunny** [comment]
As a best practice as a reusable util, I think it would be better to use `hash_equals` so it's not affected by timing attacks.

**Comment by macbre** [comment]
@Grunny - #9310

File: extensions/wikia/Security/classes/Utils.class.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Wikia\Security;
```
Comment (drsnyder) [inline extensions/wikia/Security/classes/Utils.class.php:3]: @macbre this namespace should be in `lib/Wikia/src`. We have been trying to consolidated all of the `Wikia` namespace there so its PSR-\* compliant. This also removes the need to alter the `$wgAutoloadClasses` map.

**Pull request merged**